### PR TITLE
feat: Add SPM Release GitHub Action

### PR DESCRIPTION
**Pull request by kyleleedq**
Closes [#1649](https://github.com/dequelabs/AttestIOS/issues/1649)

This commit adds the first GitHub Action to this repository and automates the process of creating an SPM Release.

This Action is manually kicked off and requires that a version/tag is provided in the semver format (2.x.x).

You can see a video of this process here:

https://github.com/dequelabs/axe-devtools-ios/assets/171582904/70a0c118-53fa-4d13-9a82-9c1f1b5c710e

### CODE REVIEW

**Review by karnett — APPROVED** [review]
LOVE.

Nice formatting of the yml file, thanks for the video also!

File: .github/workflows/spm-release.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                git push origin tag ${{ github.event.inputs.releaseVersion }}
+
+          - name: Create GitHub release
+            uses: softprops/action-gh-release@v2
+            with:
+                tag_name: ${{ github.event.inputs.releaseVersion }}
+                release_name: ${{ github.event.inputs.releaseVersion }}
+                generate_release_notes: true
+                draft: false
+                prerelease: false
```
Comment (kateowens12) [inline .github/workflows/spm-release.yml:49]: I am so grateful to have another workflow file in the repo to learn from!

**Review by kateowens12 — APPROVED** [review]
looks great!

**Pull request merged**